File: src/main/java/example/application/service/FareService.java
package example.application.service;

import example.domain.model.attempt.Attempt;
import example.domain.model.bill.Amount;
import example.domain.model.rules.AdditionalSurchargeTable;
import example.domain.model.rules.DistanceTable;
import example.domain.model.rules.FareTable;
import example.domain.model.rules.SurchargeTable;
import example.domain.model.specification.BasicFare;
import example.domain.model.specification.Destination;
import example.domain.model.specification.ExpressFare;

import org.springframework.stereotype.Service;

/**
 * 料金計算サービス
 */
@Service
public class FareService {

    FareTable fareTable;
    SurchargeTable surchargeTable;
    AdditionalSurchargeTable additionalSurchargeTable;
    DistanceTable distanceTable;

    public FareService(FareTable fareTable, SurchargeTable surchargeTable, AdditionalSurchargeTable additionalSurchargeTable, DistanceTable distanceTable) {
        this.fareTable = fareTable;
        this.surchargeTable = surchargeTable;
        this.additionalSurchargeTable = additionalSurchargeTable;
        this.distanceTable = distanceTable;
    }

    public Amount amountFor(Attempt attempt) {
        Destination to = attempt.to();
        BasicFare basic = attempt.toBasicFare(fareTable.fare(to));
        ExpressFare express  = attempt.toExpressFare(surchargeTable.surcharge(to), additionalSurchargeTable.surcharge(to));
        Amount fare = basic.calculate().add(express.calculate());
        return fare;
    }
}

File: jr-pricing/Application/Service/FareService.vb
Imports Jr.Pricing.Model
Imports Jr.Pricing.Model.Bill
Imports Jr.Pricing.Model.Rules
Imports Jr.Pricing.Model.Specification

Namespace Application.Service
    ''' <summary>
    ''' 料金計算サービス
    ''' </summary>
    ''' <remarks></remarks>
    Public Class FareService

        Private ReadOnly fareTable As FareTable
        Private ReadOnly surchargeTable As SurchargeTable
        Private ReadOnly additionalSurchargeTable As AdditionalSurchargeTable
        Private ReadOnly distanceTable As DistanceTable

        Public Sub New(fareTable As FareTable, surchargeTable As SurchargeTable, additionalSurchargeTable As AdditionalSurchargeTable, distanceTable As DistanceTable)
            Me.fareTable = fareTable
            Me.surchargeTable = surchargeTable
            Me.additionalSurchargeTable = additionalSurchargeTable
            Me.distanceTable = distanceTable
        End Sub

        Public Function AmountFor(attempt As Attempt) As Amount
            Dim [to] As Destination = attempt.To()
            Dim distance As Integer = distanceTable.GetDistance([to])
            Dim basic As BasicTickets = attempt.ToBasicTicket(fareTable.GetFare([to]), distance)
            Dim express As ExpressTickets = attempt.ToExpressTicket(surchargeTable.GetSurcharge([to]), additionalSurchargeTable.GetAdditionalSurcharge([to]), distance)
            Dim fare As Amount = basic.CalculateFare().Add(express.CalculateFare())
            Return fare
        End Function

    End Class
End Namespace